test_fatoptimizer.py

```diff
@@ -2829,6 +2829,19 @@ def f(x):
                 return 42 + 3
         ''')
 
+    def test_nested_function(self):
+        self.check_optimize('''
+            def f(x):
+                def g(x):
+                    return 100
+                return g(x) + 3
```

Comment:
Please don't try to support nested functions right now. Before trying to be the fatest, I really want to ensure that we don't break anything. A requirement for FAT Python is not break any Python application.

To inline a function, you *must* specialize the function and add guards: guard on the inline functions:
http://fatoptimizer.readthedocs.org/en/latest/fat.html#GuardFunc

and a guard on the namespace:
http://fatoptimizer.readthedocs.org/en/latest/fat.html#GuardGlobals

Later, we can discuss how to skip guards in some cases, maybe using a new configuration option.